File: 프로그래머스/lv2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    var answer = [];
//     일단 날짜 계산을 다 해야함
    let date = []
    for(let i=0;i<progresses.length;i++){
        if((100-progresses[i])%speeds[i] !==0){
            date.push(parseInt((100-progresses[i])/speeds[i])+1)
        }
        else{
            date.push(parseInt((100-progresses[i])/speeds[i]))
        }
    }
    let stack = []
    date.forEach(item=>{
        if(stack.length ===0) stack.push(item)
        else{
            if(stack[0] < item){
                answer.push(stack.length)
                stack = []
                stack.push(item)
            }
            else{
                stack.push(item)
            }
        }
    })
    if(stack.length !==0){
        answer.push(stack.length)
    }
    return answer;
}

File: 프로그래머스/lv2/42586. 기능개발/기능개발.py
from collections import deque
import math
def solution(progresses, speeds):
    answer = []
    
    #결국 앞에서부터가 중요
    #큐사용
    queue = deque(progresses)
    #카운트는 한번에 몇개 배포
    count = 0
    index = 0
    while queue:
        count = 1
        a = queue.popleft()
        time = math.ceil((100-a)/speeds[index])
        index+=1
        for k in range(len(queue)):
            queue[k] += (speeds[index+k] * time)
        while queue and queue[0] >= 100:
            queue.popleft()
            index+=1
            count+=1
        answer.append(count)
    return answer